### PR TITLE
Add additional API methods

### PR DESCRIPTION
**Pull request by delux589**
## Notes

- [Add additional API methods to help us get additional klaviyo details about a profile.](https://3.basecamp.com/3091851/buckets/9025087/todos/5319230042#__recording_5532941676)

### CODE REVIEW

File: src/services/Api.php [hunk 1]
```diff
@@ -282,6 +282,52 @@ public function profileInList($listId, $email): int
...
+     *
+     * @author	Unknown
+     * @since	v0.0.1
+     * @version	v1.0.0	Monday, November 14th, 2022.
+     * @access	public
+     * @param	mixed	$listId	
+     * @param	mixed	$email 	
+     * @return	object
+     */
+    public function getProfileFromList($listId, $email)
```
Comment (fantasticmachine) [inline src/services/Api.php:296]: Need to include Typing on this method. Indicate what type of thing the params are, and the type of thing that is being returned.

File: src/services/Api.php [hunk 2]
```diff
@@ -282,6 +282,52 @@ public function profileInList($listId, $email): int
...
+     * getProfile.
+     *
+     * @author	Unknown
+     * @since	v0.0.1
+     * @version	v1.0.0	Monday, November 14th, 2022.
+     * @access	public
+     * @param	string	$profileId	
+     * @return	object
+     */
+    public function getProfile($profileId)
```
Comment (fantasticmachine) [inline src/services/Api.php:319]: Need to include Typing on this method. Indicate what type of thing the params are, and the type of thing that is being returned.

File: src/variables/Variable.php [hunk 3]
```diff
@@ -34,6 +34,46 @@ public function lists(): mixed
...
+     *
+     * @author	Unknown
+     * @since	v0.0.1
+     * @version	v1.0.0	Monday, November 14th, 2022.
+     * @access	public
+     * @param	mixed	$listId	
+     * @param	mixed	$email 	
+     * @return	object
+     */
+    public function profileFromList($listId, $email)
```
Comment (fantasticmachine) [inline src/variables/Variable.php:48]: Need to include Typing on this method. Indicate what type of thing the params are, and the type of thing that is being returned.

File: src/variables/Variable.php [hunk 4]
```diff
@@ -34,6 +34,46 @@ public function lists(): mixed
...
+     * profileFromList.
+     *
+     * @author	Unknown
+     * @since	v0.0.1
+     * @version	v1.0.0	Monday, November 14th, 2022.
+     * @access	public
+     * @param	string	$profileId	
+     * @return	array
+     */
+    public function profile($profileId)
```
Comment (fantasticmachine) [inline src/variables/Variable.php:67]: Need to include Typing on this method. Indicate what type of thing the params are, and the type of thing that is being returned.

**Review by fantasticmachine — CHANGES_REQUESTED** [review]
We need to include the PHP Type Declarations for the new methods. See See https://www.php.net/manual/en/language.types.declarations.php

This is also going to be tricky since we have  Craft 3 and Craft 4 versions of the plugin in 2 different develop branches. I think master is the Craft 3 version so concentrate on that. 

Your changes should probably be merged into the develop-craft3 branch.
I'm not sure you'll be able to merge into develop-craft4 but the changes should also be added there too.

Sorry, it's a bit of a pain because we need 2 versions of the plugin.

**Comment by delux589** [comment]
Closing this as I created this branch off the master branch where I should have created it off develop-craft3 branch. Here is the more relevant PR.

**Pull request closed without merging**